src/compile-middleware.ts

```diff
@@ -143,7 +143,13 @@ function compileHtml(
   const document = parse5.parse(source);
   let requireScriptTag, wctScriptTag;
 
-  for (const scriptTag of dom5.queryAll(document, isJsScriptNode)) {
+  const jsNodes = dom5.queryAll(document, isJsScriptNode);
+
+  if (jsNodes.find((node) => dom5.hasAttribute(node, 'nomodule'))) {
```

Comment:
Add a brief comment? E.g. "Assume that if this document has a nomodule script, the author is already handling browsers that don't support modules, and we don't need to transform anything."

Reply:
done